lib/src/parser/recursive.dart

```diff
@@ -401,10 +363,56 @@ class RecursiveAstParser {
       tagStack.removeFirst();
     }
 
-    if (isTemplateElement) {
-      return new EmbeddedTemplateAst.parsed(
+    if (isNgContent) {
```

Comment:
I've tried the sub-typing pattern before but it still causes the issue of Element potentially being also either TemplateElement or NgContentElement - and we would just have to check those again. 

I haven't tried the factory solution yet, but will try once this issue is picked up again.